global_optimization.py

```diff
@@ -623,6 +623,7 @@ def optimize(imagefiles, lineage, realimages, synthimages, cellmaps, distmaps, w
             frame_start_temp = gerp(args.end_temp, args.start_temp, (frame_index - window_start + 1)/window)
             frame_end_temp = gerp(args.end_temp, args.start_temp, (frame_index - window_start)/window)
             temperature = gerp(frame_start_temp, frame_end_temp, current_iteration/(total_iterations))                
+            total_iterations = iteration_per_cell * lineage.count_cells_in(window_start, window_end) // window
```

Comment:
In order to ensure the total_iterations never decreases, we should take the "max" of the two above terms, or put the assignment in an "if" statement:

if(iters_per_cell * cell_count > total_iterations) total_iterations = ....

Reply:
I'll create a new pull request with that functionality